### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Vova-Stoliar**

### CODE REVIEW

**Review by nastyyvolk — APPROVED** [review]
LGTM!

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,91 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
+type Color = 'red'|'green'|'blue';
+type Shape = 'triangle'|'circle'|'rectangle';
```
Comment (VinnieJ-2k20) [inline src/figuresClasses.ts:8]:
```suggestion
type Color = 'red' | 'green' | 'blue';
type Shape = 'triangle' | 'circle' | 'rectangle';
```

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,91 @@
...
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    const first : boolean = this.a + this.b > this.c;
+    const second : boolean = this.a + this.c > this.b;
+    const third : boolean = this.b + this.c > this.a;
```
Comment (VinnieJ-2k20) [inline src/figuresClasses.ts:23]: You don't need to define types for primitive const variables.
```suggestion
    const first = this.a + this.b >= this.c;
    const second = this.a + this.c >= this.b;
    const third = this.b + this.c >= this.a;
```

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,91 @@
...
+  ) {
+    this.shape = 'circle';
+
+    if (this.a <= 0) {
+      throw Error('Error');
+    }
+  }
+
+  getArea(): number {
+    const first: number = this.a ** 2;
```
Comment (VinnieJ-2k20) [inline src/figuresClasses.ts:58]: Same here 
```suggestion
    const first = this.a ** 2;
```

**Review by VinnieJ-2k20 — CHANGES_REQUESTED** [review]
Almost done. Just fix some code style and const assignments.

File: src/figuresClasses.ts [hunk 4]
```diff
@@ -1,19 +1,91 @@
...
 
+  getArea(): number {
+    return this.a * this.b;
+  }
 }
 
-export function getInfo(figure) {
+export function getInfo(figure: Figure): string {
+  const { shape } = figure;
+  const { color } = figure;
```
Comment (VinnieJ-2k20) [inline src/figuresClasses.ts:87]:
```suggestion
  const { shape, color } = figure;
```

**Pull request closed without merging**